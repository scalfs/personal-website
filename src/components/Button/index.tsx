import { AnchorHTMLAttributes } from 'react'

type Props = AnchorHTMLAttributes<HTMLAnchorElement> & { href: string }

const Button = ({ children, className, ...props }: Props) => (
  <a
    {...props}
    target="_blank"
    className={`rounded-lg border border-slate-300 px-5 py-3 font-semibold text-blue-700 transition-colors duration-250 ease-in-out-cubic hover:border-blue-700 hover:bg-blue-100 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2 focus:ring-offset-slate-50 ${className}`}
  >
    {children}
  </a>
)

export default Button
