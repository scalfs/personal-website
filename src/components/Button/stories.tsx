import { ComponentMeta, ComponentStory } from '@storybook/react'
import Button from '.'

export default {
  title: 'Button',
  component: Button
} as ComponentMeta<typeof Button>

const Story: ComponentStory<typeof Button> = (args) => (
  <Button {...args}>Click Me</Button>
)

export const Default = Story.bind({})
