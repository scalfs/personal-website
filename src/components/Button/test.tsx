import { render, screen } from '@testing-library/react'

import Button from '.'

describe('<Button />', () => {
  it('should render the button', () => {
    const { container } = render(<Button href="">Text</Button>)

    expect(screen.getByRole('link', { name: /Text/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
