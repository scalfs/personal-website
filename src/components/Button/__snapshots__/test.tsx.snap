// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the button 1`] = `
<a
  class="rounded-lg border border-slate-300 px-5 py-3 font-semibold text-blue-700 transition-colors duration-250 ease-in-out-cubic hover:border-blue-700 hover:bg-blue-100 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2 focus:ring-offset-slate-50 undefined"
  href=""
  target="_blank"
>
  Text
</a>
`
